### PR TITLE
[5.5] Fix Collection::dump PHPDoc return type

### PR DESCRIPTION
**Pull request by ntzm**

### CODE REVIEW

File: src/Illuminate/Support/Collection.php [hunk 1]
```diff
@@ -280,7 +280,7 @@ public function dd()
     /**
      * Dump the collection.
      *
-     * @return void
+     * @return $this
```
Comment (tomschlick) [inline src/Illuminate/Support/Collection.php]: `$this` should be `self`

Comment (ntzm) [inline src/Illuminate/Support/Collection.php]: No it shouldn't, look at `each`, `forget`, `prepend`, `push`, `put`, `tap` and `transform`.

Comment (tomschlick) [inline src/Illuminate/Support/Collection.php]: Sorry about that. I haven't seen the $this keyword in a while for a return type and it looked super off. Still seems weird that they added one in with the $ while all the others don't have any special characters  😬

https://phpdoc.org/docs/latest/guides/types.html#keywords

Comment (ntzm) [inline src/Illuminate/Support/Collection.php]: No worries :D

**Pull request merged**